test/unit/tasks/cms/cms_test.rb

```diff
@@ -2,24 +2,56 @@
 
 module Tasks
   class CmsTest < ActiveSupport::TestCase
+    class CmsTestFixPaginationHref < CmsTest
+      def setup
+        @page = FactoryBot.build(:cms_partial, system_name: 'shared/pagination',
+                                 published: '<p>Title</p><a href="{{ part.link }}">Supertramp</a>')
 
-    def setup
-      @page = FactoryBot.build(:cms_partial, system_name: 'shared/pagination',
-        published: '<p>Title</p><a href="{{ part.link }}">Supertramp</a>')
+        @page.save(validate: false)
+      end
 
-      @page.save(validate: false)
+      def test_fix_pagination_href
+        assert_equal true,  @page.published.include?('{{ part.link }}')
+        assert_equal false, @page.published.include?('{{ part.url }}')
+
+        execute_rake_task 'cms/cms.rake', 'cms:fix:pagination_href'
+
+        @page.reload
+
+        assert_equal true,  @page.published.include?('{{ part.url }}')
+        assert_equal false, @page.published.include?('{{ part.link }}')
+      end
     end
 
-    def test_fix_pagination_href
-      assert_equal true,  @page.published.include?('{{ part.link }}')
-      assert_equal false, @page.published.include?('{{ part.url }}')
+    class CmsTestFixSectionEmptyTitles < CmsTest
+      def setup
+        @provider = FactoryBot.create(:provider_account)
+      end
+
+      [nil, ""].each do |empty_val|
+        define_method("test_fix_section_empty_titles_title_#{empty_val.nil? ? 'nil' : 'empty'}") do
+          section = FactoryBot.build(:cms_section, partial_path: '/', parent: @provider.sections.root, title: empty_val, system_name: 'system-name')
+          section.save!(validate: false)
+          expected_title = section.id.to_s
+
+          execute_rake_task 'cms/cms.rake', 'cms:fix:section_empty_titles'
+
+          section.reload
+          assert_equal expected_title, section.title
+        end
+      end
 
-      execute_rake_task 'cms/cms.rake', 'cms:fix:pagination_href'
+      def test_fix_section_empty_titles_valid
```

Comment:
It is not clear from then title what this test is actually testing. I _guess_ it's that the rake task do not change the titles that are valid. If so, I would name it just like so `section_empty_titles do not change existing titles`.

Reply:
I renamed it to `test_fix_section_empty_titles_valid_dont_change`. Hope it's better.

Reply:
And while nitpicking better use the `test "some behavior happens"` IMO than `def test_some_behavior_happens` :)

Reply:
I also prefer the `test "description"` over `def test_description`. I was using the second here because it's the one being used before in this file, for consistency. But I pushed it anyway.

Reply:
I think the `::test` method creates a method under the hood, but it reads better anyway.